### PR TITLE
Replaced frontend generator parser with improved version that has more semantics

### PR DESCRIPTION
**Pull request by WorldofKerry**
- Cleans up the technical debt of parsing the tree backwards (e.g. when a variable is used after assignment, don't know its type)
- Adds support for `break` and `continue`.
- Fixes comparator operator bugs
- Removed GH workflow concurrency cancel-old-jobs

### CODE REVIEW

File: python2verilog/frontend/fresh.py
Comment (Haxrox) [inline python2verilog/frontend/fresh.py:1]: Why is this file called `fresh.py`?

File: python2verilog/frontend/fresh.py [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+
+    def create_root(self):
+        """
+        Wrapper for old api
+        """
+        return self._parse_func()
+
+    def _parse_func(self, prefix: str = "_state"):
+        """
+        Parses
```
Comment (Haxrox) [inline python2verilog/frontend/fresh.py:29]: Incomplete?

File: python2verilog/ir/context.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ class Context(GenericReprAndStr):
     instances: dict[str, Instance] = field(default_factory=dict)  # generator instances
 
     @classmethod
-    def empty(cls):
+    def from_validated(cls):
         """
         Creates an empty but valid context for testing purposes
         """
```
Comment (Haxrox) [inline python2verilog/ir/context.py:82]: Is the function name accurate to what the code does? I think `def empty_valid_context` is more appropriate

File: python2verilog/frontend/fresh.py
Comment (WorldofKerry) [reply python2verilog/frontend/fresh.py:1]: Renamed

File: python2verilog/frontend/fresh.py (hunk 1 above)
Comment (WorldofKerry) [reply python2verilog/frontend/fresh.py]: Updated docs

File: python2verilog/ir/context.py (hunk 2 above)
Comment (WorldofKerry) [reply python2verilog/ir/context.py:82]: changed to `empty_valid`

File: tests/frontend/test_fresh.py
Comment (Haxrox) [inline tests/frontend/test_fresh.py:1]: Rename this too (`test_generator.py`)

**Review by Haxrox — APPROVED** [review]

**Pull request merged**